FastSimulation/ParticlePropagator/plugins/MagneticFieldMapESProducer.cc

```diff
@@ -1,30 +1,32 @@
 #include "FastSimulation/ParticlePropagator/plugins/MagneticFieldMapESProducer.h"
-#include "FastSimulation/TrackerSetup/interface/TrackerInteractionGeometryRecord.h"
-#include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
 
 #include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
 #include "FWCore/Framework/interface/ModuleFactory.h"
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
 
 #include <memory>
 
-MagneticFieldMapESProducer::MagneticFieldMapESProducer(const edm::ParameterSet& p) {
-  setWhatProduced(this);
-  _label = p.getUntrackedParameter<std::string>("trackerGeometryLabel", "");
-
+MagneticFieldMapESProducer::MagneticFieldMapESProducer(const edm::ParameterSet& p)
+    : _label(p.getUntrackedParameter<std::string>("trackerGeometryLabel", "")) {
   //    theTrackerMaterial = p.getParameter<edm::ParameterSet>("TrackerMaterial");
+
+  auto cc = setWhatProduced(this);
+  tokenGeom_ = cc.consumes(edm::ESInputTag("", _label));
+  tokenBField_ = cc.consumes();
+
+  setWhatProduced(this);
 }
 
 MagneticFieldMapESProducer::~MagneticFieldMapESProducer() {}
 
 std::unique_ptr<MagneticFieldMap> MagneticFieldMapESProducer::produce(const MagneticFieldMapRecord& iRecord) {
-  edm::ESHandle<TrackerInteractionGeometry> theInteractionGeometry;
-  edm::ESHandle<MagneticField> theMagneticField;
-
-  iRecord.getRecord<TrackerInteractionGeometryRecord>().get(_label, theInteractionGeometry);
-  iRecord.getRecord<IdealMagneticFieldRecord>().get(theMagneticField);
+  const edm::ESHandle<TrackerInteractionGeometry>& theInteractionGeometry =
+      iRecord.getRecord<TrackerInteractionGeometryRecord>().getHandle(tokenGeom_);
+  const edm::ESHandle<MagneticField>& theMagneticField =
+      iRecord.getRecord<IdealMagneticFieldRecord>().getHandle(tokenBField_);
```

Comment:
If the handle is not needed, `getData` can be preferred
By the way, is it needed/convenient to retrieve these infos once per event?